### ISSUE TITLE
Test aliasing sul-streaming links

### ISSUE THREAD

**Issue by hannahfrost** [sul-dlss/media#38]
Towards the goal of migrating off of sul-streaming (old wowza): we need to test if we can alias the links to content on NEW wowza that has been migrated from OLD wowza so that the published links out in the world can gracefully resolve to the new server.

If this works, then we can move off the old box more quickly, and then develop a plan for updating the links over time.

Task: 
- [ ] move some sample files from sul-streaming to the new wowza server, preserving the folder structure
- [ ] test alias from old url to new url

**Comment by ndushay** [comment]
talk to devops re:  the aliasing piece.  We know we can do directories in wowza.

rights issues for any of this content?

**Comment by hannahfrost** [comment]
@ndushay 99% of the content on sul-streaming is unrestricted access. There is a small tranche of material that is Stanford-only (we hacked a way to support that in Drupal)

**Comment by LynnMcRae** [comment]
Some hints on the streaming part syntax of the sul-streaming URL with this view of the page source serving one.  Do we need to rehost the page that presents the flash player as well, since it shares a host name with the streaming?

https://sul-streaming.stanford.edu/collections/hershman/war/antoni_2006.html, an iframe within https://lib.stanford.edu/women-art-revolution/janine-antoni

```
<body><div id="player1"><a href="http://www.macromedia.com/go/getflashplayer">Get the Flash Player</a> to see this player.</div>
<script type="text/javascript">
        var s1 = new SWFObject( '../../mediaplayer_new/player.swf', 'mpl', '480', '360', '8' );
        s1.addParam( 'allowfullscreen', 'true' );
        s1.addVariable( 'streamer', 'rtmp://sul-streaming.stanford.edu/simplevideostreaming' ); // <-- change to rtmp url
        s1.addVariable( 'file', 'hershman/war/M1639_Antoni_2006_web' ); // <-- add id variable and set to flv filename (minus .flv extension)
        s1.addVariable( 'image', '/collections/hershman/war/images/M1639_Antoni_2006.png' );
        s1.addVariable( 'width', '480' );
        s1.addVariable( 'height', '360' );
        s1.write( 'player1' );
    </script></body></html>
```

**Comment by LynnMcRae** [comment]
A Bucky Fuller example ... https://collections.stanford.edu/bucky/bin/mediaplayer?fileID=157138371X&n=1

Here there is no iframe intermediate source of the html... from View Source

```
<table cellpadding="0" cellspacing="0" align="left" style="padding: 0px 0 20px 0">
  <tr>
    <td align="right">
      <script type="text/javascript" src="/bucky/js/swfobject.js"></script>
      <div id="player" style="padding-bottom: 3px;"/>
      <script type="text/javascript">
        var s1 = new SWFObject("/bucky/swf/mediaplayer.swf","mediaplayer","480","70","8");
        s1.addParam("allowfullscreen","true");
        s1.addParam("bgcolor", "#ede8dd");
        s1.addVariable("autostart", "true");
        s1.addVariable("file","rtmp://sul-streaming.stanford.edu/simplevideostreaming"); 
        s1.addVariable("id","mp3:bucky/ss1_002_002f-a"); 
        //s1.addVariable("image","BuckyVideoSplash.png");
        s1.addVariable("width","480");
        s1.addVariable("height","70");
        s1.addVariable("showeq","true");
        s1.write("player");
      </script>

      <noscript>
        <embed 
          src="/bucky/swf/mediaplayer.swf" 
          width="480"
          height="70"
          allowscriptaccess="always"
          allowfullscreen="true"
          flashvars="file=rtmp://sul-streaming.stanford.edu/simplevideostreaming&autostart=true&id=mp3:bucky/ss1_002_002f-a&image=BuckyVideoSplash.png"
        />
      </noscript>
    </td>
  </tr>
```

**Comment by ndushay** [comment]
Spoke to Erin at lunch today.   If the urls are all to the _viewer_ code, which also lives on the sul-streaming box, then it's more complex than setting up the aliasing for the actual _content_ on sul-streaming to the same content copied to our new wowza box  (sul-mediaserver-prod).  

Our initial take was:
- it is feasible to put content on the new wowza box and do a regex proxy with our proxy service for the content
- it is mucho desirable to change any of OUR urls pointing to sul-streaming to point directly to the new content
- if the old links are to a _viewer_, then more work is needed, and it may be better to just accession the stuff properly.

@hannahfrost: are the urls Lynn provided appropriate examples of:
- what urls are used to get to sul-streaming content (like, as a user)?
  - https://sul-streaming.stanford.edu/collections/hershman/war/antoni_2006.html
  - are they to a streaming _viewer_ or to the actual _content_?  The above implies both.
- where are the sul-streaming urls used?  (which ones are used where?   _viewer_ vs. \* content*)
  - https://collections.stanford.edu
  - https://lib.stanford.edu/women-art-revolution/janine-antoni
  - where else?  Do we care about urls we don't control (e.g. user bookmarks?)?

**Comment by LynnMcRae** [comment]
I believe the URLs to the actual content are the rmtp://sul-streaming ones embedded in the presentation of the player you get from the https calls. If so, only those rmtp calls might need to be redirected to a new host. That would not affect how collection.stanford.edu presents the player or any bookmarks users might have.  But since I can only report net behavior in the browser, I can't answer how the two might be linked underneath.

**Comment by ndushay** [comment]
where does the _viewer_ live?   Seems like it lives on sul-streaming??

**Comment by hannahfrost** [comment]
@ndushay Yes, everything for the older streaming content lives on sul-streaming
A user is presented with the content via a HTML page, e.g., https://sul-streaming.stanford.edu/collections/hershman/war/antoni_2006.html

And the URLs are used several places. I have a document that addresses that. See: https://docs.google.com/spreadsheets/d/1xOcpTQm44h_2KdZURsmijmdCsxyqI4YHC23PZnMXm2Q/edit#gid=0

**Comment by ndushay** [comment]
This has been closed in favor of #50.   This ticket was really about assessing the viability of the approach.  It was a little more complex than just aliasing the streaming content, but we believe the approach outlined in #50 is probably viable.
